### PR TITLE
chore: submenu refactor for modules IA changes

### PR DESCRIPTION
**Pull request by ashit-rath**
## Description
PR refactors the submenu for query creation to pull out the content Ui to a separate component for reuse

#### PR fixes following issue(s)
EE PR https://github.com/appsmithorg/appsmith-ee/pull/3188

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
	- Improved the Editor's file explorer by separating the menu content into a new component for better code organization.

- **New Features**
	- Introduced a search feature within the file explorer to filter file operations quickly.

- **Documentation**
	- Added type definitions and comments for the new `ExplorerMenuContent` component and its props for clarity.

### CODE REVIEW

File: app/client/src/pages/Editor/Explorer/Files/Submenu.tsx [hunk 1]
```diff
@@ -59,24 +149,13 @@
...
     setQuery("");
   }, [show]);
 
-  const onChange = useCallback((value) => {
-    setQuery(value);
-  }, []);
-
   const handleOpenChange = (open: boolean) => {
     if (open) {
       // handle open
```
[the remark this answers is not in the document]
Comment (ashit-rath) [reply app/client/src/pages/Editor/Explorer/Files/Submenu.tsx:161]: It already has `openMenu` in the dependency array

**Review by ankitakinger — APPROVED** [review]
LGTM

**Pull request merged**